### ISSUE TITLE
App bugging out with simple closures

### ISSUE THREAD

**Issue by RayKay91** [lukehaas/RunJS#193]
First off, love this app it's been invaluable to me in my journey of web dev I really appreciate it.

The issue I had was upon updating from 1.11 to the latest release the following code no longer outputs correctly in the console.

```
function outer() {
    let counter = 0

    return function inner() {
        counter++

        if (counter <= 3) {
            console.log(counter)
        } else {
            console.log(`Sorry, function use is maxed out`)
        }
    }
}

const myNewFunc = outer()

myNewFunc()
myNewFunc()
myNewFunc()
myNewFunc()
```
What the result should be is:

```
1
2
3
'Sorry, function use is maxed out'
```

I redownloaded 1.11 and checked the code and it worked properly and cross checked it also with the code runner extension in VSCode and also a Node REPL in my Terminal app (I use Hyper).

Instead in version 1.12 and 1.12.1 of RunJS I get 

```
1
'Sorry, function use is maxed out'
'Sorry, function use is maxed out'
'Sorry, function use is maxed out'
```

So I tried logging the counter itself, and the number value changes depending upon the number of times the `myNewFunc()` is invoked.

When I invoke it 2 times I get 
```
1
3
```

when invoked 4 times I get

```
1
5
6
7
```

**Comment by lukehaas** [comment]
@RayKay91 thanks for raising this. You're right, this is a defect that was introduced in version 1.12.
This issue is related to the runtime values feature, which was re-worked in 1.12.
I'll aim to release a fix before the end of the month.

In the meantime, you can avoid this issue by using 1.11 or by turning off runtime values in the advanced preferences:
![Screenshot 2020-11-05 at 09 28 00](https://user-images.githubusercontent.com/741344/98223472-0eb95500-1f4a-11eb-8503-c06f6b832d41.png)

**Comment by lukehaas** [comment]
@RayKay91 this is now fixed in RunJS version 1.12.2. Thanks again for raising it.
